### PR TITLE
Adding VMware vSphere Disk Mode support (Dependent, Independent - P…

### PR DESCRIPTION
**Pull request by sbernhard**
fixes #18815 - Adding VMware vSphere Disk Mode support

### CODE REVIEW

**Comment by ohadlevy** [comment]
@BernhardAtix thanks! FYI - please note that #3979 potentially rewrite this part of the form

**Comment by ohadlevy** [comment]
ok to test

File: app/views/compute_resources_vms/form/vmware/_volume.html.erb [hunk 1]
```diff
@@ -23,3 +23,17 @@
            :label => _("Eager zero"), :label_size => "col-md-2", :disabled => !new_host},
            "true",
            "false" %>
+<%
+# regarding to the following documentation:
```
Comment (timogoebel) [inline app/views/compute_resources_vms/form/vmware/_volume.html.erb]: Could you please remove these comments? They don't belong in Foreman's source.

File: app/views/compute_resources_vms/form/vmware/_volume.html.erb [hunk 2]
```diff
@@ -23,3 +23,17 @@
...
+# http://pubs.vmware.com/vsphere-65/index.jsp?topic=%2Fcom.vmware.wssdk.apiref.doc%2Fvim.vm.device.VirtualDiskOption.DiskMode.html
+# there are some more disk modes available (nonpersistent, undoable, append) but as far as I can see, 
+# these disk modes don't work and don't exist on current vmware vsphere configuration GUI.
+#  ['nonpersistent', :nonpersistent], 
+#  ['undoable', :undoable],
+#  ['append', :append]]
+%>
+<% disk_mode_values = [['Dependent', :persistent],
```
Comment (timogoebel) [inline app/views/compute_resources_vms/form/vmware/_volume.html.erb]: Please move these options to vmware's compute resource model.

`firmware_types` is one example for this.
https://github.com/theforeman/foreman/blob/develop/app/models/compute_resources/foreman/model/vmware.rb#L170

Please note, that you need to mark the names as translatable, e.g. `_('Dependent')`.

File: app/views/compute_resources_vms/form/vmware/_volume.html.erb [hunk 3]
```diff
@@ -23,3 +23,17 @@
...
+   ['Independent - Persistent', :independent_persistent],
+   ['Independent - Nonpersistent', :independent_nonpersistent]] %>
+<%= selectable_f f, :mode, disk_mode_values, { }, :class => "span5", :label => _("Disk mode"), :label_size => "col-md-2", :disabled => !new_host %>
+
```
Comment (timogoebel) [inline app/views/compute_resources_vms/form/vmware/_volume.html.erb]: Nitpick: Please remote the empty line.

**Review by timogoebel — CHANGES_REQUESTED** [review]
@BernhardAtix : Thanks for your PR. I left some comments inline. Could you also please provide a Redmine issue for this?

**Comment by sbernhard** [comment]
fixes #18815 
=> added redmine issue for it

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 4]
```diff
@@ -175,6 +175,14 @@ def firmware_types
       }
     end
 
+    def disk_mode_types
+      {
+        "persistent" => N_("Persistent"),
```
Comment (timogoebel) [inline app/models/compute_resources/foreman/model/vmware.rb]: This isn't translated correctly.
`N_()` just marks a string for translation but doesn't actually translate it.
Please use either `_()` or translate it in the view.
```ruby
<%= select_f f, :mode, compute_resource.disk_mode_types, :first, ->(d) { _(d.last) }, { }, :class => "span5", :label => _("Disk mode"), :label_size => "col-md-2", :disabled => !new_host %>
```

**Review by timogoebel — CHANGES_REQUESTED** [review]
@BernhardAtix : Thanks for the updates. I left one small comment inline.

**Review by timogoebel — APPROVED** [review]
Thanks @BernhardAtix. Works well.

**Pull request merged**